src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -118,47 +118,62 @@ internal sealed class BuildServerConnection
             var clientDir = buildPaths.ClientDirectory;
             var timeoutNewProcess = timeoutOverride ?? TimeOutMsNewProcess;
             var timeoutExistingProcess = timeoutOverride ?? TimeOutMsExistingProcess;
-            var clientMutexName = GetClientMutexName(pipeName);
             Task<NamedPipeClientStream> pipeTask = null;
-            using (var clientMutex = new Mutex(initiallyOwned: true,
-                                               name: clientMutexName,
-                                               createdNew: out var holdsMutex))
+            Mutex clientMutex = null;
+            var holdsMutex = false;
+            try
             {
                 try
                 {
-                    if (!holdsMutex)
+                    var clientMutexName = GetClientMutexName(pipeName);
+                    clientMutex = new Mutex(initiallyOwned: true, name: clientMutexName, out holdsMutex);
```

Comment:
Unfortunately I don't have any good ideas on how to test this. No way I can think of to force Mutex ctor to throw an exception here.

Reply:
Why not pass in a mutex indirection?  In standard builds, pass in a simple version that just instantiates and returns "new Mutexes".  In a test, pass in a version that throws.  That's one way we simulate exceptions in IDE tests.

Reply:
:bulb: I prefer to force the real error to occur by pre-creating a mutex with the same name and using ACLs that will prevent this constructor from accessing it.

Reply:
@sharwell like that idea. Have an example I can look at. Been a while since I've done ACLs.

Reply:
This passes for me on Windows.

```csharp
[Fact]
public void TestMutexACL()
{
    var user = Environment.UserDomainName + "\\" + Environment.UserName;

    using (var outer = new Mutex(true, "SomeName", out bool createdNew))
    {
        Assert.True(createdNew);
        var mutexSecurity = outer.GetAccessControl();
        mutexSecurity.AddAccessRule(new MutexAccessRule(user, MutexRights.FullControl, AccessControlType.Deny));
        outer.SetAccessControl(mutexSecurity);

        Assert.Throws<UnauthorizedAccessException>(() => new Mutex(true, "SomeName", out createdNew));
    }
}
```

Reply:
Looks like the easiest way to make the Mutex constructor throw is to just pass in a name longer that 260 characters.